cmd/localstack/xraydaemon.go

```diff
@@ -133,8 +133,9 @@ func initDaemon(config *cfg.Config) *Daemon {
 	awsConfig, session := conn.GetAWSConfigSession(&conn.Conn{}, config, config.RoleARN, config.Region, noMetadata)
 	log.Infof("Using region: %v", aws.StringValue(awsConfig.Region))
 
-	log.Debugf("ARN of the AWS resource running the daemon: %v", config.ResourceARN)
-	telemetry.Init(awsConfig, session, config.ResourceARN, noMetadata)
+	if enableTelemetry {
+		telemetry.Init(awsConfig, session, config.ResourceARN, noMetadata)
```

Comment:
This would initialize `telemetry.T` which is why the additional nil checks were added below